services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaController.kt

```diff
@@ -53,14 +53,44 @@ class HankeKayttajaController(
             ]
     )
     @PreAuthorize("@hankeKayttajaAuthorizer.authorizeHankeTunnus(#hankeTunnus, 'VIEW')")
-    fun whoami(@PathVariable hankeTunnus: String): WhoamiResponse {
+    fun whoami(@PathVariable hankeTunnus: String): WhoAmIDto {
         val userId = currentUserId()
 
         val hankeIdentifier = hankeService.findIdentifier(hankeTunnus)!!
         val permission = permissionService.findPermission(hankeIdentifier.id, userId)!!
 
         val hankeKayttaja = hankeKayttajaService.getKayttajaByUserId(hankeIdentifier.id, userId)
-        return WhoamiResponse(hankeKayttaja?.id, permission.kayttooikeustasoEntity)
+        return WhoAmIDto(hankeKayttaja?.id, permission.kayttooikeustasoEntity)
+    }
+
+    @GetMapping("hankkeet/my-permissions")
+    @Operation(
+        summary = "Get your permissions for your own projects",
+        description = "Returns a map of current users Hanke identifiers and respective permissions."
+    )
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(
+                    description = "Permissions grouped by hankeTunnus.",
+                    responseCode = "200",
+                ),
+                ApiResponse(description = "Access denied.", responseCode = "401")
```

Comment:
This response (along with 500 internal server error) is automatically documented on every endpoint. It doesn't need to be repeated here.

Reply:
Ok. Removed.